components/core/src/service.rs

```diff
@@ -83,7 +81,89 @@ impl FromStr for BindingMode {
     }
 }
 
-////////////////////////////////////////////////////////////////////////
+/// A binding from a service name to a service group that provides that service
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct ServiceBind {
+    name: String,
+    service_group: ServiceGroup,
+}
+
+impl ServiceBind {
+    pub fn new<S: Into<ServiceGroup>>(name: &str, service_group: S) -> Self {
+        Self {
+            name: name.to_string(),
+            service_group: service_group.into(),
+        }
+    }
+
+    pub fn name(&self) -> &String {
```

Comment:
```suggestion
    pub fn name(&self) → &str {
```
`&String` is rarely needed since `&str` is more flexible.